### ISSUE TITLE
When `DEFAULT_EDGE_LIMIT` set to 3, some graph algorithm extensions fail for graphs with edge count >= `3 * N + 1`

### ISSUE THREAD

**Issue by wbkboyer** [graph-algorithms/edge-addition-planarity-suite#64]
Observed during #62 that release build of `planarity` failed for commands `d34` (i.e. drawing, `K_{3,3}`, and `K_4` extensions) on `.g6` input files with `m >= 34` for `n = 11`, i.e. with edge-count greater than or equal to `3 * N + 1`:

```
Start testing all graphs in "E:\graphs\11\test.n11.m34.g6".
Unable to interpret bits on line 1 to populate adjacency matrix.
Unable to read graph on line 1 from .g6 read iterator.

Encountered error while running command 'd' on all graphs in "E:\graphs\11\test.n11.m34.g6".
...
Start testing all graphs in "E:\graphs\11\test.n11.m34.g6".
Unable to interpret bits on line 1 to populate adjacency matrix.
Unable to read graph on line 1 from .g6 read iterator.

Encountered error while running command '4' on all graphs in "E:\graphs\11\test.n11.m34.g6".
...
Start testing all graphs in "E:\graphs\11\test.n11.m34.g6".
Unable to interpret bits on line 1 to populate adjacency matrix.
Unable to read graph on line 1 from .g6 read iterator.

Encountered error while running command '3' on all graphs in "E:\graphs\11\test.n11.m34.g6".
```

When one sets `#define DEFAULT_EDGE_LIMIT      6` in `graphStructures.h`, execution proceeds without issue.

**Comment by wbkboyer** [comment]
TL;DR When we `_decodeGraph` we call `gp_DynamicAddEdge()`, which tries to `gp_EnsureArcCapacity()` to a new larger arc capacity after `gp_AddEdge()` failed the first time. However, if you have previously attached the extensions for draw planar, `K_4` search, and `K_{3, 3}` search to the graph (as is done in `testAllGraphs()` regardless of the command being run), the function table is overridden so that instead of `graphUtils.c`'s `_EnsureArcCapacity()`, we call:
* `_DrawPlanar_EnsureArcCapacity()`
* `_K33Search_EnsureArcCapacity()`
* `_K4SearchEnsureArcCapacity()`
which all return `NOTOK` with the comment:
> The current implementation does not support an increase of arc (edge record) capacity once the extension is attached to the graph data structure.  This is only due to not being necessary to support.
> For now, it is easy to ensure the correct capacity before attaching the extension, but support could be added later if there is some reason to do so.

One solution would entail calling `AttachAlgorithm()` after we `readGraphUsingG6ReadIterator()` for the first time (i.e. when `pG6ReadIterator->numGraphsRead == 1`) and before we call `gp_Embed()`, since that'll mean that we'll have already resized the `ArcCapacity` (which should be preserved when we `gp_ReinitializeGraph()`)

<details>
<summary>Investigation details</summary>

To simplify and expedite the investigation, I figured out that the smallest graph order for which this issue would occur is `N = 8`: this is the first order for which the threshold for when we'd need to resize the `ArcCapacity` (which is set when we `gp_InitGraph()` during the `beginG6ReadIterationFromG6FilePointer()` flow to `2 * DEFAULT_EDGE_LIMIT * 8 = 2 * 3 * 8 = 48`, corresponding to 24 edges) is less than the max number of edges we'd need to be able to represent (i.e. `(8 * 7) / 2 = 28`). That meant that I was able to use the `test.n8.m25.g6` file to reproduce the issue.

_COMING SOON_
</details>

**Comment by wbkboyer** [comment]
The initial proposed solution to only `AttachAlgorithm()` to the `pG6ReadIterator->currGraph` and `copyOfOrigGraph` `graph` structures after we have `readGraphUsingG6ReadIterator()` for the first time works when we are guaranteed to have only *one* number of edges for all graphs in a file, but will fail if we have multiple `M` sizes (e.g. we generate all 10 vertex graphs and stick them in one `.g6` file)

There are two alternate solutions:
1. Before we `AttachAlgorithm()` in `testAllGraphs()`, we could `gp_EnsureArcCapacity()` to `(N * (N - 1))`
2. For every iteration of the loop, after processing the current graph, `gp_Free()` the old graphs and create brand new graphs to assign to the `G6ReadIterator` and `copyOfOrigGraph`

Testing needs to be done to see the speed impact of either of these choices.
